primal-bit/src/lib.rs

```diff
@@ -171,11 +170,21 @@ impl BitVec {
     /// }
     /// ```
     pub fn from_elem(nbits: usize, bit: bool) -> BitVec {
+        use std::ptr;
+
         let nblocks = nbits.checked_add(BITS - 1).expect("capacity overflow") / BITS;
+        let mut out_vec = Vec::with_capacity(nblocks);
+
+        unsafe {
+            out_vec.set_len(nblocks);
+            ptr::write_bytes(out_vec.as_mut_ptr(), if bit { !0 } else { 0 }, nblocks);
+        }
```

Comment:
Did you consider something like `vec![if bit { !0 } else { 0 }; nblocks]`?  Its implementation is fairly specialized, so hopefully still fast, and then we won't need unsafe code just for initialization.